services/databaseEncryptionService.ts

```diff
@@ -0,0 +1,163 @@
+    throw error;
+  }
+
+  if (isCancellationError(error)) {
+    throw new EncryptionError(ERROR_CODES.USER_CANCELLED, 'Authentication was cancelled');
+  }
+
+  throw error;
+}
+
+async function createNewKey(): Promise<Uint8Array> {
+  const key = generateRandomKey();
+  await SecureStore.setItemAsync(SECURE_STORE_KEYS.ENCRYPTION_KEY, base64.fromByteArray(key), { requireAuthentication: false });
+  await setStoredEncryptionMode('basic');
+  return key;
+}
+
+async function loadExistingKey(): Promise<Uint8Array> {
+  const mode = await getStoredEncryptionMode();
+  const keyBase64 = await SecureStore.getItemAsync(SECURE_STORE_KEYS.ENCRYPTION_KEY, { requireAuthentication: mode === 'protected' });
+  
+  if (!keyBase64) {
+    throw new EncryptionError(ERROR_CODES.KEY_NOT_FOUND, 'Encryption key is missing. Your data cannot be decrypted.');
+  }
+
+  return base64.toByteArray(keyBase64);
+}
+
+export async function initializeEncryption(): Promise<void> {
+  if (keyCache) {
+    return;
+  }
+
+  if (initializationPromise) {
+    return initializationPromise;
+  }
+
+  initializationPromise = (async () => {
+    try {
+      keyCache = await loadExistingKey();
+    } catch (error) {
+      if (error instanceof EncryptionError && error.code === ERROR_CODES.KEY_NOT_FOUND) {
+        keyCache = await createNewKey();
+      } else {
+        throw classifyError(error);
+      }
+    } finally {
+      initializationPromise = null;
+    }
+  })();
+
+  return initializationPromise;
+}
+
+async function getStoredEncryptionMode(): Promise<EncryptionMode> {
+  try {
+    const mode = await AsyncStorage.getItem(ASYNC_STORAGE_KEYS.ENCRYPTION_MODE);
+    return mode === 'protected' ? 'protected' : 'basic';
+  } catch {
+    return 'basic';
+  }
+}
+
+async function setStoredEncryptionMode(mode: EncryptionMode): Promise<void> {
+  await AsyncStorage.setItem(ASYNC_STORAGE_KEYS.ENCRYPTION_MODE, mode);
+}
+
+export function getEncryptionKey(): Uint8Array {
+  if (!keyCache) {
+    throw new Error('Encryption key not initialized. Call initializeEncryption first.');
+  }
+  return keyCache;
+}
+
+export async function getEncryptionMode(): Promise<EncryptionMode> {
+  try {
+    return await getStoredEncryptionMode();
+  } catch (error) {
+    console.error('Error getting encryption mode:', error);
+    return 'basic';
+  }
+}
+
+export async function reWrapKEK(requireAuth: boolean): Promise<void> {
+  if (!keyCache) {
+    throw new Error('Encryption key not initialized.');
+  }
+
+  const newMode: EncryptionMode = requireAuth ? 'protected' : 'basic';
+  
+  try {
+    await SecureStore.setItemAsync(
+      SECURE_STORE_KEYS.ENCRYPTION_KEY, 
+      base64.fromByteArray(keyCache), 
+      { requireAuthentication: requireAuth }
+    );
+    await setStoredEncryptionMode(newMode);
+  } catch (error) {
+    throw classifyError(error);
+  }
+}
+
+export function clearKeyCache(): void {
+  keyCache = null;
+}
+
+export async function clearAllKeys(): Promise<void> {
```

Comment:
This name doesn't convey the implications.

```suggestion
export async function deleteEncryptionKey(): Promise<void> {
```